packages/versioning/test/versioned-dependencies.test.ts

```diff
@@ -905,6 +884,32 @@ describe("versioning: dependencies", () => {
     const [v1] = runProjections(runner.program, MyService);
     ok(v1.projectedTypes.get(MyService));
   });
+
+  it("have a service namespace that indirectly depends on multiple versioned libraries", async () => {
+    const { MyService } = (await runner.compile(`
+        @service({title: "Test"})
+        @useDependency(Azure.ResourceManager.Versions.v1)
+        @test("MyService")
+        namespace MyOrg.MyService {
+        }
+        
+        @versioned(Versions)
+        namespace Azure.Core {
+          enum Versions { v1: "v1" }
+        }
+        
+        @versioned(Versions)
+        namespace Azure.ResourceManager { 
+          enum Versions { 
+            @useDependency(Azure.Core.Versions.v1)
+            v1: "v1" 
+          }
+        }
+      `)) as { MyService: Namespace };
+
+    const [v1] = runProjections(runner.program, MyService);
+    ok(v1.projectedTypes.get(MyService));
```

Comment:
This test case _is_ the repro, but it passes. The diagnostics get raised during the `runProjections` step, but the projected `MyService` in fact includes all relevant types. Ideally I would catch those diagnostics here so this can serve as an effective regression test. Not sure why they get swallowed.

Reply:
Something would have to check the diagnostics on the projected program. I added this assert, and this + two other tests fail: https://github.com/tjprescott/cadl/compare/versioning/EnumUnrevert...nguerrera:typespec:versioning/EnumUnrevert